langchain4j-mariadb/src/main/java/dev/langchain4j/store/embedding/mariadb/MariaDbFilterMapper.java

```diff
@@ -0,0 +1,117 @@
+package dev.langchain4j.store.embedding.mariadb;
+
+import static java.lang.String.format;
+
+import dev.langchain4j.store.embedding.filter.Filter;
+import dev.langchain4j.store.embedding.filter.comparison.*;
+import dev.langchain4j.store.embedding.filter.logical.And;
+import dev.langchain4j.store.embedding.filter.logical.Not;
+import dev.langchain4j.store.embedding.filter.logical.Or;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+abstract class MariaDbFilterMapper {
+
+    public String map(Filter filter) {
+        if (filter instanceof IsEqualTo) {
+            return mapEqual((IsEqualTo) filter);
```

Comment:
Here and in other places it might be concise to use new syntax (Pattern Matching for instanceof): `if (filter instanceof IsEqualTo isEqualFilter) { mapEqual(isEqualFilter) }`

OpenRewrite might be able to [fix](https://docs.openrewrite.org/running-recipes/popular-recipe-guides/migrate-to-java-17) it automatically

Reply:
About pattern matching, current project compatibility is java 17, only langchain4j-jlama use java 21. It might be a bit early for that ?